### PR TITLE
Add Bifoldable and Bitraverse instances for Of

### PR DESCRIPTION
**Pull request by torgeirsh**

### CODE REVIEW

File: src/Data/Functor/Of.hs [hunk 1]
```diff
@@ -49,6 +53,16 @@ instance Bifunctor Of where
   {-#INLINE second #-}
 #endif
 
+#if MIN_VERSION_base(4,10,0)
+instance Bifoldable Of where
+  bifoldMap f g ~(a :> b) = f a <> g b
+  {-#INLINE bifoldMap #-}
+
```
Comment (treeowl) [inline src/Data/Functor/Of.hs]: Is this extra laziness actually useful in practice?

**Comment by chessai** [comment]
CI failing because of Semigroup-Monoid compat issues on 8.2.2

File: src/Data/Functor/Of.hs (hunk 1 above)
Comment (torgeirsh) [reply src/Data/Functor/Of.hs]: Actually I don't know. It's there in the instances for (,), so I thought it best to err on the side of caution.

**Comment by torgeirsh** [comment]
> CI failing because of Semigroup-Monoid compat issues on 8.2.2

What's the best resolution? Replace <> with \`mappend\`?

**Comment by torgeirsh** [comment]
@treeowl Can this be merged, or should I remove the lazy pattern match?

**Comment by chessai** [comment]
@torgeirsh im in favour of removing it. if you do, i will merge this

**Comment by chessai** [comment]
Forgot to mention, 2 things:

1. Add a changelog instance, please?
2. Add `@since` pragmas to the instances?

**Comment by torgeirsh** [comment]
Should I add a new version 0.2.4.0 to the changelog? I don't know what @since is, and couldn't find anything about it in the repository or web search. Could you please elaborate what you mean?

**Comment by treeowl** [comment]
It's a Haddock annotation.

```haskell
-- | Some documentation!
--
-- @since 0.1.1
foo :: a -> b -> c
foo = baz

-- | @since 1.2.3
instance Blah Bloop where ...
```

**Comment by chessai** [comment]
Nice, thanks!

**Pull request merged**